lib/rucio/core/did_meta_plugins/elasticsearch_meta.py

```diff
@@ -0,0 +1,406 @@
+        """
+        doc_id = f"{scope.internal}{name}"
+        try:
+            # First, get the current document
+            doc = self.client.get(index=self.index, id=doc_id)
+
+            # Check if the key exists in the document
+            if key in doc['_source']:
+                # Use script to remove the field
+                script = {
+                    "script": {
+                        "source": f"ctx._source.remove('{key}')",
+                        "lang": "painless"
+                    }
+                }
+                self.client.update(index=self.index, id=doc_id, body=script)
+        except elastic_exceptions.NotFoundError as err:
+            raise exception.DataIdentifierNotFound(f"No metadata found for DID '{scope}:{name}' not found") from err
+        except Exception as err:
+            raise exception.RucioException(err)
+
+    def list_dids(
+        self,
+        scope: "InternalScope",
+        filters: Union[list[dict[str, Any]], dict[str, Any]],
+        did_type: str = 'collection',
+        ignore_case: bool = False,
+        limit: "Optional[int]" = None,
+        offset: "Optional[int]" = None,
+        long: bool = False,
+        recursive: bool = False,
+        ignore_dids: "Optional[list]" = None,
+        *,
+        session: "Optional[Session]" = None
+    ) -> "Iterator[dict[str, Any]]":
+        """
+        List DIDs (Data Identifier).
+
+        :param scope: The scope of the DIDs to search.
+        :param filters: The filters to apply to the DID search.
+        :param did_type: The type of DID (default is 'collection').
+        :param ignore_case: Whether to ignore case (default is False).
+        :param limit: The maximum number of DIDs to return.
+        :param offset: The starting point for the search (used for pagination).
+        :param long: Whether to return extended information (scope, name, did_type, bytes, length) (default is False).
+        :param recursive: Whether to search recursively (currently unsupported).
+        :param ignore_dids: A list of DIDs to ignore (default is an empty list).
+        :param session: The database session in use.
+        :returns: A generator yielding DIDs as strings (when `long` is False) or dictionaries (when `long` is True).
+        :raises UnsupportedOperation: If recursive searches are requested (currently unsupported).
+        :raises RucioException: If an error occurs during the search.
+        """
+
+        if not ignore_dids:
+            ignore_dids = []
+
+        # backwards compatability for filters as single {}.
+        if isinstance(filters, dict):
+            filters = [filters]
+
+        # Create Elasticsearch query
+        fe = FilterEngine(filters, model_class=None, strict_coerce=False)
+        elastic_query_str = fe.create_elastic_query(
+            additional_filters=[
+                ('scope', operator.eq, str(scope.external)),
+                ('vo', operator.eq, str(scope.vo))
```

Comment:
```suggestion
                ('vo', operator.eq, scope.vo)
```

`InternalScope.vo` is always a `str`